Symmetric Tree.java

```diff
@@ -0,0 +1,81 @@
+// Time Complexity :o(n)
+// Space Complexity : worst case o(n) or o(height of tree)
+// Did this code successfully run on Leetcode : yes
+// Any problem you faced while coding this : RECURSIVE
+ I faced issue in line 27 
+ if(nodeA.val !=nodeB.val || nodeA==null || nodeB==null)
+ If I run above condition it is throwing run time error . 
+ How is above code different from current line 27
+
+// Any problem you faced while coding this : ITERATIVE
+
+If I change my order of if conditions it is throwing run time error.
+
+        if(a.val!=b.val){
+    return false;
+    }
+if(a ==null && b ==null) continue;
+if(a ==null || b ==null){
+    return false;
+    }

```

Comment:
Always the base conditions have to be on top, so that if you don't do redundant code execution further. 

I believe this is the reason.